modules/aggregation-layers/src/cpu-grid-layer/cpu-grid-layer.js

```diff
@@ -81,44 +78,42 @@ export default class CPUGridLayer extends CompositeLayer {
   }
 
   updateState({oldProps, props, changeFlags}) {
-    this.updateGetValueFuncs(oldProps, props);
+    this.updateGetValueFuncs(oldProps, props, changeFlags);
     const reprojectNeeded = this.needsReProjectPoints(oldProps, props, changeFlags);
 
     if (changeFlags.dataChanged || reprojectNeeded) {
       // project data into hexagons, and get sortedBins
       this.getLayerData();
     } else {
-      const dimensionChanges = this.getDimensionChanges(oldProps, props) || [];
+      const dimensionChanges = this.getDimensionChanges(oldProps, props, changeFlags) || [];
       dimensionChanges.forEach(f => typeof f === 'function' && f.apply(this));
     }
   }
 
-  colorElevationPropsChanged(oldProps, props) {
-    let colorChanged = false;
-    let elevationChanged = false;
-    for (const p of COLOR_PROPS) {
-      if (oldProps[p] !== props[p]) {
-        colorChanged = true;
-      }
-    }
-    for (const p of ELEVATION_PROPS) {
-      if (oldProps[p] !== props[p]) {
-        elevationChanged = true;
-      }
-    }
-    return {colorChanged, elevationChanged};
-  }
-
-  updateGetValueFuncs(oldProps, props) {
+  updateGetValueFuncs(oldProps, props, changeFlags) {
+    const {getFillColor, getElevation} = this.getDimensionUpdaters();
     let {getColorValue, getElevationValue} = props;
     const {colorAggregation, getColorWeight, elevationAggregation, getElevationWeight} = this.props;
-    const {colorChanged, elevationChanged} = this.colorElevationPropsChanged(oldProps, props);
 
-    if (colorChanged && getColorValue === null) {
+    const getColorValueChanged = this.needUpdateDimensionStep(
+      getFillColor[0],
```

Comment:
Should this be a map instead of an array?

Reply:
Made it a map inside cpuAggregationManager